src/lib.rs

```diff
@@ -65,11 +69,34 @@ impl Passage {
     ///     "kid": "..."
     ///   }"#;
     ///
-    /// // If you are using an Element, the Passage authentication JWT will be sent to your application via a cookie with the key `psg_auth_token`
+    /// let passage = Passage::new(String::from(app_id), String::from(api_key), String::from(pub_key));
+    /// ```
+    pub fn new(app_id: String, api_key: String, pub_jwk: String) -> Self {
```

Comment:
This is a weird halfway state to me.

Originally, I didn't need any Passage APIs and was fine with just a static public key. But I knew whenever I (or anyone who discovered my little package 😄) needed to use the API, I'd need to add `api_key` and actual HTTP requests, at which point it would make sense to just dynamically look up the `pub_jwk` rather than requiring it to be passed here. (You had mentioned not being interested in this dynamic lookup to avoid a network failure on startup, but the lookup would actually be lazy, whenever the first request happens that needs the pub key.)

However, as of this MR, we still haven't defined what API requests to Passage will look like (request lib, etc.), so we still have to ask for `pub_jwk`. But that means when we implement dynamic lookup in the future, we would probably want to change this API _again_ to not require `pub_jwk` to be passed. 

To avoid that, maybe something like this would be even better:

```rust
let passage = Passage::new(app_id, api_key).set_pub_jwk(pub_jwk);
```

And the `Passage` struct would change to `pub_jwk: Option<String>`. When we add dynamic lookup, consumers can just drop the `set_pub_jwk` call, or keep it, either way, but the API won't change.

We can add an AuthError variant that indicates the public key is unavailable if it's `None`. In the future we'd repurpose this a lookup request error.